### ISSUE TITLE
`initial_population` not effectively used/retained for multiobjective problems?

### ISSUE THREAD

**Issue by kaurao** [ahmedfgad/GeneticAlgorithmPython#279]
Hi,

Thanks for PyGAD, it's a great resource.
I am trying to use it for an multiobjective optimization task. I have a good idea for what good solutions could be so I am providing them via the `initial_population` argument. However, some of those solutions are bot being used, I think.

My set up is as follows.
```
ga_instance = pygad.GA(num_genes=68,
                       sol_per_pop=100,
                       initial_population=initial_population,
                       num_generations=100,
                       num_parents_mating=np.round(sol_per_pop/2).astype(int),
                       parent_selection_type=parent_selection_type,
                       gene_space={'low': 0, 'high': 1},
                       crossover_type="uniform",
                       mutation_type=None,
                       mutation_num_genes=[5, 1],
                       keep_elitism=10,
                       fitness_func=fitness_func,
                       on_generation=on_generation)
```

The image shows the final front 0 as blue dots and the red cross as one of the initial solutions that I provided whiuch apparently "disappeared". 

![image](https://github.com/ahmedfgad/GeneticAlgorithmPython/assets/10744124/bc07c0d4-802b-4b1a-b8d4-78744d21b5b6)

I will appreciate any tips on how to set this up properly using PyGAD, thanks!

**Comment by kaurao** [comment]
My guess is the difference is coming from `parent_selection_type="tournament_nsga2"` (the details and the figure above) versus `parent_selection_type="nsga2"` which seems to work as expected (see the figure below).

![image](https://github.com/ahmedfgad/GeneticAlgorithmPython/assets/10744124/329a67a4-d209-4677-9282-990a5f9f1717)

**Comment by ahmedfgad** [comment]
`tournament_nsga2` and `nsga2` are different types of parent selectors for multi-objective optimization using NSGA-II.

`tournament_nsga2` applies a tournament between the candidates and parents are selected randomly. This means the solution with the best fitness is not guaranteed to be selected. In contrast, `nsga2` always select the best solutions as parents.

**Comment by kaurao** [comment]
Makes sense. Can you please provide referene(s) for `tournament_nsga2` I have not come across it before. Thanks!

**Comment by ahmedfgad** [comment]
These are some resources that describes tournament selection for NSGA-II:
* https://www.thearmchaircritic.org/mansplainings/nsga-ii-non-dominated-sorting-genetic-algorithm-ii
* https://arxiv.org/abs/2203.11550

You can definitely have more resources.
